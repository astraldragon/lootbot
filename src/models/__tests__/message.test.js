const message = require("../message");

describe("models/message", () => {
  beforeEach(() => {
    this.model = {
      belongsTo: jest.fn("belongsTo").mockReturnValue()
    };

    this.models = {
      Tier: jest.fn("Tier")
    };

    this.sequelize = {
      define: jest.fn("define").mockReturnValue(this.model)
    };

    this.dataTypes = {
      STRING: "string",
      TEXT: "test",
      ENUM: jest.fn((...args) => args.join(",")),
      INTEGER: "integer"
    };
  });

  it("should create the model definition", () => {
    message(this.sequelize, this.dataTypes);
    expect(this.sequelize.define).toMatchSnapshot();
  });

  it("should associate loot with tiers", () => {
    const Message = message(this.sequelize, this.dataTypes);
    Message.associate(this.models);

    expect(Message.belongsTo).toHaveBeenCalledWith(this.models.Tier, {
      foreignKey: "tier_id"
    });
  });
});
