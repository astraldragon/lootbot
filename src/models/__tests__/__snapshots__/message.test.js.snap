// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`models/message should create the model definition 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "Message",
      Object {
        "delay": "integer",
        "guild": "string",
        "message": "test",
        "name": "string",
        "tier_id": "integer",
        "type": "draw,tier,reward",
        "user_id": "string",
      },
      Object {
        "paranoid": true,
      },
    ],
  ],
}
`;
